src/Cms/User.php

```diff
@@ -450,6 +461,8 @@ public function logout($session = null): void
             // privilege change
             $session->regenerateToken();
         }
+        
+        $kirby->trigger('user.logout:after', $old);
```

Comment:
I think we should also pass the session here if it still exists (in the `else` case) for consistency:

```php
if ($session->data()->get() === []) {
    // session is now empty, we might as well destroy it
    $session->destroy();
    
    $kirby->trigger('user.logout:after', $old, null);
} else {
    // privilege change
    $session->regenerateToken();
    
    $kirby->trigger('user.logout:after', $old, $session);
}
```

Also: Why the `hardcopy`? The user object itself doesn't change during logout, does it?

Reply:
I also didn't get it into if clause because the session object still exists when the session destroy.

You were right. I thought that the user object would disappear when the user session ended.

Reply:
Your new implementation passes the destroyed session object to the hook anyway – but it's not possible to write anything to a destroyed session anymore, so passing it will be confusing for hook users who want to do something with the session still.

Reply:
So no more need to send the `session` after user logged out?

Reply:
No, as I wrote above, I‘d pass the session in the second case (where it still exists), but not in the first case.

Reply:
I get it now. So that it can use the session when there is a privilege change.
But i cant implement to test nullable session with stable solution, so test failed about that :(( 
Can you fix this?